File: Python/Iniciante/1045.py
import math
a, b, c = map(float, input().split())

ordena = [a, b, c]
ordena.sort()

A = ordena[2]
B = ordena[1]
C = ordena[0]

triangulo = True

if A >= B+C:
    print("NAO FORMA TRIANGULO")
    triangulo = False

if triangulo is True:
    if math.pow(A, 2) == math.pow(B, 2) + math.pow(C, 2): print("TRIANGULO RETANGULO")
    if math.pow(A, 2) > math.pow(B, 2) + math.pow(C, 2): print("TRIANGULO OBTUSANGULO")
    if math.pow(A, 2) < math.pow(B, 2) + math.pow(C, 2): print("TRIANGULO ACUTANGULO")

    contagem_elementos = {}
    for i in ordena:
        contagem = contagem_elementos.get(i, 0)
        contagem_elementos[i] = contagem + 1

    for elemento, contagem in contagem_elementos.items():
        if contagem == 3: print("TRIANGULO EQUILATERO")
        if contagem == 2: print("TRIANGULO ISOSCELES")

File: Java/Iniciante/1045.java
import java.io.IOException;
import java.util.Scanner;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {

      Scanner input = new Scanner(System.in);
  
      double n1 = input.nextDouble();
      double n2 = input.nextDouble();
      double n3 = input.nextDouble();
      double[] valores = {n1, n2, n3};
      Arrays.sort(valores);
  
      double a = valores[2]; // Maior lado
      double b = valores[1]; // Segundo maior
      double c = valores[0]; // Menor lado
  
      if (a >= b + c) {
        System.out.println("NAO FORMA TRIANGULO");
      } 
      else {
        if (Math.pow(a, 2) == Math.pow(b, 2) + Math.pow(c, 2)) {
          System.out.println("TRIANGULO RETANGULO");
        } else if (Math.pow(a, 2) > Math.pow(b, 2) + Math.pow(c, 2)) {
          System.out.println("TRIANGULO OBTUSANGULO");
        } else if (Math.pow(a, 2) < Math.pow(b, 2) + Math.pow(c, 2)) {
          System.out.println("TRIANGULO ACUTANGULO");
        }
  
        if (a == b && b == c) {
          System.out.println("TRIANGULO EQUILATERO");
        } else if (a == b || b == c || a == c) {
          System.out.println("TRIANGULO ISOSCELES");
      }
    }
  }
}
